### ISSUE TITLE
Unfriendly error message, if 1st phase of commit fails in a segment.

### ISSUE THREAD

**Issue by hlinnaka** [greenplum-db/gpdb#4530]
This is now easy to reproduce with deferred unique constraints, but in principle the same could happen with any error that is raised in the prepare phase:

```
CREATE TABLE unique_tbl (i int UNIQUE DEFERRABLE, t text);
INSERT INTO unique_tbl VALUES (1, 'one');

BEGIN;
SET CONSTRAINTS ALL DEFERRED;
INSERT INTO unique_tbl VALUES (1, 'one');
COMMIT;
```
This fails with error at COMMIT:
```
WARNING:  the distributed transaction 'Abort [Prepared]' broadcast failed to one or more segments for gid = 1518178709-0000001209.  Retrying ... try 0
NOTICE:  Releasing segworker groups to retry broadcast.
ERROR:  The distributed transaction 'Prepare' broadcast failed to one or more segments for gid = 1518178709-0000001209. (cdbtm.c:692)
```

That's not very user-friendly. In upstream, you get:
```
ERROR:  duplicate key value violates unique constraint "unique_tbl_i_key"
DETAIL:  Key (i)=(1) already exists.
```

**Comment by hlinnaka** [comment]
There's a GPDB_90_MERGE_FIXME comment about this at: https://github.com/greenplum-db/gpdb/blob/master/src/test/regress/input/constraints.source#L356

**Comment by hlinnaka** [comment]
Fixed by commit b3c50e4086.
